Add feature to save turnip data from JSON to the manager

patrikhillner opened the pull request:
Also add feature usage instructions to index page

patrikhillner committed:
Add feature to save turnip data from JSON to the manager

Also add feature usage instructions to index page

patrikhillner commented:
Still needs to update a variable name: `checked` -> `autoUpdateJson`

patrikhillner committed:
Refactor variable name

* Refactor checked and setChecked to jsonAutoUpdate and setJsonAutoUpdate respectively. This is to improve clarity on what it does.
* Remove redundant semi-colon.

patrikhillner committed:
Refactor turnip JSON to use only one textarea

Also set jsonAutoUpdate state to be tracked in TurnipPriceStore.

Pingviinituutti commented:
Updating the JSON does not work? For example, changi the textarea value to `[]` does nothing.

patrikhillner commented:
This PR adds feature to fill blank spaces with data. Saving does not overwrite all data, so it works as it should.

patrikhillner committed:
Fix JSON text area updating only when ticking checkbox

JSON text area did not update automatically when ticked. This was due to the fact that jsonAutoUpdate variable did not extend the observable property that the similarly named variable in the store did. This commit fixes that by directly using the variable (through reference) of the store and updating only that through the method in the store.

patrikhillner committed:
Add isNaN check for stored price when saving JSON text area content

If user had removed a price through the TurnipPriceDialog, and then wanted to add a price through the JSON text area for the same (morning/afternoon) date, then stored price was `NaN`. This caused undefined and null checks to return false, thus not saving the new price.

patrikhillner committed:
Fix calendar tiles not updating after saving JSON text area

Since calendar tiles do not directly use the same objects as in turnips list in the turnip store, the calendar tiles did not update when saving JSON text area and null prices were set to a non-null value. Add check to see if the tile needs to update its turnip.
Existing TurnipPriceStore functions refresh turnip objects in TurnipPriceDialog when turnip list in store is updated, causing the object to re-render and do the check.

patrikhillner commented:
Final issues of this PR have been fixed:
* Issue: JSON text area updating only when ticking checkbox.
* Add isNaN check for stored price when saving JSON text area content.
* Issue: Calendar tiles with null values did not update after saving JSON text area containing data for respective tiles.
